client/src/hooks/permissions/index.ts

```diff
@@ -0,0 +1,25 @@
+import { useProfile } from 'hooks/profile';
+
+import type { Permission, RoleName } from './enums';
+
+export function usePermissions() {
+  const { data, isLoading } = useProfile();
+
+  const roles: RoleName[] = [];
+  const permissions: Permission[] = [];
+
+  data?.roles?.forEach((role) => {
+    roles.push(role.name);
+    role.permissions?.forEach((permission) => permissions.push(permission));
+  });
+
+  const hasRole = (role: RoleName) => {
+    return !isLoading && !!roles?.includes(role);
+  };
```

Comment:
Take in account that unless you want this specifically to check for a admin role, this wouldn't have much sense, as we could not know what a role 'can do'. Unless for the time being you just want to discriminate between admin and user

Reply:
Yes, we know. It will be there just in case we need for any reason. We will base the front end on the permissions instead.